File: app/src/main/java/abdullah/elamien/bloodbank/models/Shoutout.java
package abdullah.elamien.bloodbank.models;

import android.widget.ImageView;

import androidx.databinding.BindingAdapter;

import com.bumptech.glide.Glide;
import com.google.firebase.Timestamp;

import java.text.SimpleDateFormat;
import java.util.Date;

public class Shoutout {
    private String name, email, phone, specificAddress, description, city, bloodType, image;
    private Timestamp date;

    public Shoutout() {
    }

    @BindingAdapter("profileImage")
    public static void loadShoutoutImage(ImageView view, String imageUrl) {
        Glide.with(view.getContext())
                .load(imageUrl)
                .into(view);
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getSpecificAddress() {
        return specificAddress;
    }

    public void setSpecificAddress(String specificAddress) {
        this.specificAddress = specificAddress;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getBloodType() {
        return bloodType;
    }

    public void setBloodType(String bloodType) {
        this.bloodType = bloodType;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getDate() {
        Date timePosted = date.toDate();
        SimpleDateFormat df = new SimpleDateFormat("dd MMM yyyy");
        return df.format(timePosted);
    }

    public void setDate(Timestamp date) {
        this.date = date;
    }
}

File: app/src/main/java/abdullah/elamien/bloodbank/models/Shoutout.kt
package abdullah.elamien.bloodbank.models

import android.widget.ImageView
import androidx.databinding.BindingAdapter
import com.bumptech.glide.Glide
import com.google.firebase.Timestamp
import java.text.SimpleDateFormat

class Shoutout {
    var name: String? = null
    var email: String? = null
    var phone: String? = null
    var specificAddress: String? = null
    var description: String? = null
    var city: String? = null
    var bloodType: String? = null
    var image: String? = null
    private var date: Timestamp? = null

    fun getDate(): String {
        val timePosted = date!!.toDate()
        val df = SimpleDateFormat("dd MMM yyyy")
        return df.format(timePosted)
    }

    fun setDate(date: Timestamp) {
        this.date = date
    }

    companion object {
        @BindingAdapter("imageUrl")
        @JvmStatic
        fun loadShoutoutImage(view: ImageView, imageUrl: String) {
            Glide.with(view.context).load(imageUrl)
                    .into(view)
        }
    }
}
